From e936af33fa4c7a475076d287c723901f1408c888 Mon Sep 17 00:00:00 2001
From: Yuan Juntao <juntao.yuan@intel.com>
Date: Thu, 29 Jan 2015 18:18:10 +0800
Subject: [PATCH] mmc: host: optimize mmc runtime PM initialization process.

With this change, mmc host controller can be put into D3 when
there is no device connected to the host.

Change-Id: I63ead8e92186e86fcc8df508db97d1b432be81a7
Signed-off-by: Yuan Juntao <juntao.yuan@intel.com>
---
 drivers/mmc/core/host.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/mmc/core/host.c b/drivers/mmc/core/host.c
index 77d8704..eb1d039 100644
--- a/drivers/mmc/core/host.c
+++ b/drivers/mmc/core/host.c
@@ -504,13 +504,6 @@ struct mmc_host *mmc_alloc_host(int extra, struct device *dev)
 	host->max_blk_size = 512;
 	host->max_blk_count = PAGE_CACHE_SIZE / 512;
 
-	pm_runtime_set_active(&host->class_dev);
-	pm_runtime_enable(&host->class_dev);
-	/*
-	 * ignore the children by default
-	 */
-	pm_suspend_ignore_children(&host->class_dev, true);
-
 	return host;
 
 free:
@@ -546,6 +539,12 @@ int mmc_add_host(struct mmc_host *host)
 #endif
 	mmc_host_clk_sysfs_init(host);
 
+	/*
+	 * ignore the children by default
+	 */
+	pm_suspend_ignore_children(&host->class_dev, true);
+	pm_runtime_enable(&host->class_dev);
+
 	mmc_start_host(host);
 	if (!(host->pm_flags & MMC_PM_IGNORE_PM_NOTIFY))
 		register_pm_notifier(&host->pm_notify);
-- 
1.9.1

